bukuserver/forms.py

```diff
@@ -1,26 +1,49 @@
 """Forms module."""
 # pylint: disable=too-few-public-methods, missing-docstring
-import wtforms
 from flask_wtf import FlaskForm
+from wtforms import BooleanField, FieldList, StringField, TextAreaField, HiddenField
+from wtforms.validators import DataRequired, InputRequired, ValidationError
+from buku import DELIM
 
 
 class SearchBookmarksForm(FlaskForm):
-    keywords = wtforms.FieldList(wtforms.StringField('Keywords'), min_entries=1)
-    all_keywords = wtforms.BooleanField('Match all keywords')
-    deep = wtforms.BooleanField('Deep search')
-    regex = wtforms.BooleanField('Regex')
+    keywords = FieldList(StringField('Keywords'), min_entries=1)
+    all_keywords = BooleanField('Match all keywords')
+    deep = BooleanField('Deep search')
+    regex = BooleanField('Regex')
 
 
 class HomeForm(SearchBookmarksForm):
-    keyword = wtforms.StringField('Keyword')
+    keyword = StringField('Keyword')
 
 
 class BookmarkForm(FlaskForm):
-    url = wtforms.StringField('Url', name='link', validators=[wtforms.validators.InputRequired()])
-    title = wtforms.StringField()
-    tags = wtforms.StringField()
-    description = wtforms.TextAreaField()
-    fetch = wtforms.HiddenField(filters=[bool])
-
-class ApiBookmarkForm(BookmarkForm):
-    url = wtforms.StringField(validators=[wtforms.validators.DataRequired()])
+    url = StringField('Url', name='link', validators=[InputRequired()])
+    title = StringField()
+    tags = StringField()
+    description = TextAreaField()
+    fetch = HiddenField(filters=[bool])
+
+
+def delim_not_in_tag(form, field):
```

Comment:
`valid_tag`?

Reply:
I think a valid tag would be one that checks all validators.

Reply:
A "valid tag" is a _tag_ value which is _valid_. As of now, this is defined as "not containing a comma"; but "valid tag" is a higher-level concept and thus should be preferred to "not containing a comma" when identifying tag validation.

Reply:
What i mean is that `delim_not_in_tag` is a validator method that checks for a specific condition related to delim, and raises a message accordingly. If criteria were to change in the future I think one could easily  add/remove other validators from the tags validators lists.
Btw I think it could also be stored in `validators.py` and called `delim_not_in_field` instead.  

Would you prefer something like `def custom_tag_validator(form, field)` instead?

Reply:
What I'm saying that you _shouldn't_ have a validator that "checks for a specific condition" (a low-level concept). This condition is only used for validating _tags_, but there's no good reason for any code outside of the validator itself to know what exact conditions a _tag_ must match to be _valid_.

What you should have is _one_ validator, that ensures validity of the value as a tag (a high-level concept), and being the only part of the code that is concerned with what exactly that constitutes.

As for naming, I'm following the general trend of validator naming here. Except this one isn't in PascalCase, I guess; I'm uncertain if it should be applied. (…Also, I'd say it should be moved to the top of the file, instead of randomly popping up somewhere in the middle.)